cmd/wait/runner.go

```diff
@@ -199,5 +199,68 @@ func (r *runner) run(ctx context.Context, _ *cobra.Command, _ []string) error {
 		r.logger.LogCtx(ctx, "message", "CoreDNS is ready")
 	}
 
+	// We wait for external-dns for Azure and AWS. This prevents failures where
+	// the CNCF suit is started before external-dns is functional.
+	// This is not necessary on KVM as there is no external-dns app running.
+	if r.flag.Provider != "kvm" {
+		r.logger.LogCtx(ctx, "message", "waiting for external-dns to be ready")
+
+		targetLabels := map[string]string{
+			"giantswarm.io/service-type": "managed",
+			"app.kubernetes.io/name":     "external-dns",
+		}
+
+		o := func() error {
+			serviceLabelSelector := labelsToSelector(targetLabels)
+			services, err := k8sClient.CoreV1().Services("kube-system").List(ctx, v1.ListOptions{
+				LabelSelector: serviceLabelSelector,
+			})
+			if err != nil {
+				r.logger.LogCtx(ctx, "message", "error listing services", "error", err)
+				return microerror.Mask(err)
+			}
+			if len(services.Items) == 0 {
+				message := fmt.Sprintf("external-dns service not found using label selector %#q", serviceLabelSelector)
+				r.logger.LogCtx(ctx, "message", message)
+				return microerror.Mask(notReadyError)
+			}
+
+			service := services.Items[0]
```

Comment:
What happens / could there be multiple?

Reply:
No - something would be very wrong if there are multiple. We do the same with coreDNS - so I assumed this to be safe.